roles/mcpserver-model/templates/model.service.yaml.j2

```diff
@@ -0,0 +1,56 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: "{{ ansible_operator_meta.name }}-api"
+  namespace: "{{ ansible_operator_meta.namespace }}"
+  labels:
+    {{ lookup("template", "../common/templates/labels/common.yaml.j2")  | indent(width=4) | trim }}
+    app.kubernetes.io/component: '{{ deployment_type }}-api'
+{% if service_annotations %}
+  annotations:
+    {{ service_annotations | indent(width=4) }}
+{% endif %}
+spec:
+  ports:
+{% if service_type | lower == "nodeport" %}
+    - port: 8000
+      protocol: TCP
+      targetPort: 8000
+      name: http
+{% if nodeport_port is defined %}
+      nodePort: {{ nodeport_port }}
+{% endif %}
+{% elif service_type | lower != 'loadbalancer' and loadbalancer_protocol | lower != 'https' %}
+    - port: 8000
+      protocol: TCP
+      targetPort: 8000
+      name: http
+{% endif %}
+{% if ingress_type | lower == 'route' and route_tls_termination_mechanism | lower == 'passthrough' %}
+    - port: 443
+      protocol: TCP
+      targetPort: 8853 # TODO: Use something like 8053 and update nginx.con
+      name: https
+{% endif %}
+{% if service_type | lower == 'loadbalancer' and loadbalancer_protocol | lower == 'https' %}
+    - port: {{ loadbalancer_port }}
+      protocol: TCP
+      targetPort: 8000
+      name: https
+{% elif service_type | lower == 'loadbalancer' and loadbalancer_protocol | lower != 'https' %}
```

Comment:
duplicate code in these two if blocks, what really changes between the two is 'name: (http|https)'

Reply:
@rcarrillocruz Thanks for your comment. I agree that we should be able to simplify these if blocks, but these were copied from the existing code under `roles/model` and I do not want to change the logic in this PR as I have not written any test cases yet. Can we push to the next sprint?

Reply:
wfm, it's not that is wrong, just a refactor